### ISSUE TITLE
Side menu on holiday app doesn't meet contrast reqs

### ISSUE THREAD

**Issue by adriannelee** [cds-snc/design-gc-conception#2087]
Note: This is an issue specific to the holiday app implementation of the side nav and doesn't concern the main component. Hence the low priority.

ITAO audit result:
Location: Canada Holidays
Date Tested: 2025-12-05
Status: Fail
Severity: Serious
WCAG Criterion: 1.4.11 Non-Text Contrast (Level AA)
Impact: Users with visual impairments may struggle to distinguish UI states due to insufficient colour contrast.

Description:
Non-text interactive elements do not meet the minimum 3:1 colour contrast ratio required by WCAG.

Example 1: In the sidebar, selected items such as “About this app” or “Submit a holiday” highlight in #D7E5F5 vs #F1F2F3 (1.14:1).

Example 2: In the "Holidays" section, the year selector uses #E1E1E1 on #FFFFFF (1.31:1).

Recommendation:
Adjust the colour palette to ensure all non-text portions meet or exceed a 3:1 contrast ratio.

<img width="1225" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBKzFkREE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--cd4b59ed29971966ddfd819a0574ac099f0d43a4/image.png" alt="image.png" />

**Comment by ethanWallace** [comment]
We will need design to review if the hover state of the nav link and group components meet wcag compliance.

**Comment by melaniebmn** [comment]
@adorayi should we create a duplicate ticket for dev once the design has been reviewed / done?

**Comment by nmakuch** [comment]
#### Example 1: We've documented this in another ticket:

"After doing some reading / research it seems like we're actually ok on these backgrounds or boundaries not hitting 3:1 contrast.

In the linked W3 article (https://www.w3.org/WAI/WCAG22/Understanding/non-text-contrast.html)
specifically the section on Boundaries:

"This success criterion does not require that controls have a visual boundary indicating the hit area, but if the visual indicator of the control is the only way to identify the control, then that indicator must have sufficient contrast. If text (or an icon) within a button or placeholder text inside a text input is visible and there is no visual indication of the hit area then the success criterion is passed. If a button with text also has a colored border, since the border does not provide the only indication there is no contrast requirement beyond the text contrast"

Because the main indicator here is the text bolding / underline being removed on the text and pointer interactions being disabled, the colour of the background isn't vital to success.

#### Example 2: We've documented this in another ticket:

"In terms of the first screenshot, the fill is not the only cue here it's working in combination with the cursor switching to a pointer, which is a pretty common pattern and even shows up in some of the most accessible design systems two I've noticed using a similar gray hover state are:

Atlassian in their side nav and in their select component (https://atlassian.design/components/select/examples)
France in their side menu (https://www.systeme-de-design.gouv.fr/version-courante/fr)"

**Comment by nmakuch** [comment]
I'm suggesting no change to the things flagged above, but we could improve the contrast a bit by removing the background fill on the sidebar, it's not necessary but it's a quick fix if we feel up to it:
https://www.figma.com/design/672zNYNzI3NqotZCccZjXT/Holidays-app?node-id=0-1&p=f&t=gPviLiDoQPekMkdG-0
